code/modules/mob/living/silicon/silicon_defense.dm

```diff
@@ -97,11 +97,18 @@
 		return
 	return ..()
 
-/mob/living/silicon/electrocute_act(shock_damage, obj/source, siemens_coeff = 1, safety = 0, tesla_shock = 0, illusion = 0, stun = TRUE)
+/mob/living/silicon/electrocute_act(shock_damage, obj/source, siemens_coeff = 1, safety = 0, tesla_shock = 0, illusion = 0, stun = TRUE, gib = FALSE)
 	if(buckled_mobs)
 		for(var/mob/living/M in buckled_mobs)
 			unbuckle_mob(M)
-			M.electrocute_act(shock_damage/100, source, siemens_coeff, safety, tesla_shock, illusion, stun)	//Hard metal shell conducts!
+			M.electrocute_act(shock_damage/100, source, siemens_coeff, safety, tesla_shock, illusion, stun, gib)	//Hard metal shell conducts!
+	if(istype(src, /mob/living/silicon/robot) && gib)	
+		var/mob/living/silicon/robot/bot = src
```

Comment:
We should never be checking for src's type, if we want this to be only on cyborg then it should be on ``/mob/living/silicon/robot/electrocute_act``.